scipio/src/dma_file.rs

```diff
@@ -285,6 +285,48 @@ impl DmaFile {
         Reactor::get().alloc_dma_buffer(size)
     }
 
+    /// Returns the system's preferred alignment for the file
+    async fn fetch_device_alignment(&self) -> io::Result<u64> {
+        use std::fs;
+        use std::path;
+
+        // dev major + minor gives enough information about te file to determine physical location
+        let stats = self.statx().await?;
+        let major = stats.stx_dev_major;
+        let minor = stats.stx_dev_minor;
+
+        // /sys/dev/block/major:minor is a symlink to the device in /sys/devices/
+        let dir_path = format!("/sys/dev/block/{}:{}", major, minor);
```

Comment:
Please note that the paths are different if you are dealing with an lvm or raid device. Take a look at how the seastar implementation of `iotune` handles that. If we discover that the device is an array we need to jump from there to its constituent devices

Reply:
https://github.com/scylladb/seastar/blob/master/apps/iotune/iotune.cc#L109 ?

Reply:
Yes but note that this PR is a bit outdated.

After it was started, I have merged support for parsing sysfs for other reasons. It shouldn't be too hard at this point to do the rest.